build-logic/integration-testing/src/main/kotlin/gradlebuild.cross-version-tests.gradle.kts

```diff
@@ -46,15 +46,30 @@ fun configureTestFixturesForCrossVersionTests() {
         }
     }
 }
+
 val releasedVersions = gradleModule.identity.releasedVersions.orNull
 
+fun Test.addShadedJarDependency() {
+    val moduleIdentity = gradleModule.identity
+    val shadedJarFile: Provider<String> = moduleIdentity.baseName.zip(moduleIdentity.version) { baseName, version ->
+        "shaded-jar/gradle-tooling-api-shaded-${version.baseVersion.version}.jar"
+    }
+    val jarFile = project(":tooling-api").layout.buildDirectory.file(shadedJarFile)
```

Comment:
This doesn't seem right to me. We should be declaring a dependency on the tooling api project, and the tooling api project should have a configuration that advertises its shaded tooling API jar

Reply:
@jvandort PTAL at the update